app/src/main/java/com/weather/weathercheckerapp/module/MainActivityModule.kt

```diff
@@ -0,0 +1,15 @@
+package com.weather.weathercheckerapp.module
+
+import com.weather.weathercheckerapp.viewmodel.MainViewModel
+import dagger.Module
+import dagger.Provides
+
+@Module
+class MainActivityModule {
+
+    @Provides
+    internal fun provideMainViewModel(): MainViewModel {
```

Comment:
I don't think you need this provides function.
MainViewModel already has a @Inject constructor, dagger should be able to inject that without this function.

Actually I don't think you need this class at all.